src/Type.cpp

```diff
@@ -317,6 +331,8 @@ Type::Type(vector<const Type*>& struct_fields, bool isStruct, bool packed,
     else
         eType = eUnion;
     sid =  sequence++;
+	struct_attr_flag = false;
```

Comment:
You should not need this.

Reply:
I need this to identify whether given type is struct or not. One should not generate type attributes to other variables.

Reply:
Can you use existing fields to distinguish `struct` and `enum`?